### PR TITLE
GH-148 GH-151 and untested block post

### PR DESCRIPTION
**Pull request by velzevur**
GH-148 - get block by height
GH-151 - get block by hash

Untested function for posting a block

### CODE REVIEW

File: apps/aecore/src/aec_blocks.erl [hunk 1]
```diff
@@ -116,15 +117,22 @@ serialize_for_network(B = #block{}) ->
     {ok, jsx:encode(serialize_to_map(B))}.
 
 serialize_to_map(B = #block{}) ->
+    Pow =
+        case is_integer(B#block.pow_evidence) of
```
Comment (lucafavatella) [inline apps/aecore/src/aec_blocks.erl]: I expect this to fail always as that field does not seem integer. I expect this to be list of 42 integers.

Refs:
* https://github.com/aeternity/epoch/wiki/Block-Header
* https://github.com/aeternity/epoch/blob/1fa7a978ddaa897e2bb8cfdd948f2cc1e0e8aac9/apps/aecore/c_src/aec_pow_cuckoo_nif.cpp#L55
* https://github.com/aeternity/epoch/blob/1fa7a978ddaa897e2bb8cfdd948f2cc1e0e8aac9/apps/aecore/test/aec_pow_cuckoo_tests.erl#L60

File: config/swagger.yaml [hunk 2]
```diff
@@ -123,15 +147,23 @@ definitions:
       version:
         type: integer
         format: int64
-  Header:
+      pow:
+        type: string
+      txs:
+        type: string
```
Comment (lucafavatella) [inline config/swagger.yaml]: This type looks wrong. Reference https://github.com/aeternity/epoch/wiki/Block
I expect list of transactions.

File: apps/aehttp/priv/swagger.json [hunk 3]
```diff
@@ -59,31 +59,64 @@
...
             "schema" : {
               "$ref" : "#/definitions/Block"
             }
           }
         },
         "security" : [ ]
-      },
+      }
+    },
+    "/block-by-hash" : {
```
Comment (sennui) [inline apps/aehttp/priv/swagger.json]: This is probably is subjective, but it doesn't look right to have block-by-hash and block-by-height. block/height/XX and block/hash/YY - also not perfect. block/ZZ and header for a type probably the cleanest.

Not a blocker but keep in mind when desigining the whole API

**Comment by velzevur** [comment]
moved to #168

File: apps/aehttp/priv/swagger.json (hunk 3 above)
Comment (velzevur) [reply apps/aehttp/priv/swagger.json]: I didn't like it either; got inspiration from testnet

**Pull request closed without merging**